applications/StructuralMechanicsApplication/tests/cpp_tests/test_high_cycle_fatigue_constitutive_law_3d.cpp

```diff
@@ -0,0 +1,155 @@
+    NodeType::Pointer p_node_2 = test_model_part.CreateNewNode(2, 1.0, 0.0, 0.0);
+    NodeType::Pointer p_node_3 = test_model_part.CreateNewNode(3, 0.0, 1.0, 0.0);
+    NodeType::Pointer p_node_4 = test_model_part.CreateNewNode(4, 0.0, 0.0, 1.0);
+
+    Tetrahedra3D4<NodeType> Geom = Tetrahedra3D4<NodeType>(p_node_1, p_node_2, p_node_3, p_node_4);
+
+    stress_vector = ZeroVector(6);
+
+    strain_vector = ZeroVector(6);
+    strain_vector[0] = 5.0e-03;
+    strain_vector[1] = 0.0;
+    strain_vector[2] = 0.0;
+    strain_vector[3] = 0.0;
+    strain_vector[4] = 0.0;
+    strain_vector[5] = 0.0;
+
+    material_properties.SetValue(DENSITY, 7850.0);
+    material_properties.SetValue(YOUNG_MODULUS, 2.01e11);
+    material_properties.SetValue(POISSON_RATIO, 0.3);
+    material_properties.SetValue(FRACTURE_ENERGY, 20e5);
+    material_properties.SetValue(YIELD_STRESS, 838.9e6);
+    material_properties.SetValue(SOFTENING_TYPE, 1);
+    Vector high_cycle_fatigue_parameters(7);
+    high_cycle_fatigue_parameters[0]=0.5;
+    high_cycle_fatigue_parameters[1]=0.7;
+    high_cycle_fatigue_parameters[2]=0.5;
+    high_cycle_fatigue_parameters[3]=0.0068;
+    high_cycle_fatigue_parameters[4]=3.35;
+    high_cycle_fatigue_parameters[5]=0.0133;
+    high_cycle_fatigue_parameters[6]=0.0068;
+    material_properties.SetValue(HIGH_CYCLE_FATIGUE_PARAMETERS, high_cycle_fatigue_parameters);
+
+    // Set constitutive law flags:
+    Flags& ConstitutiveLawOptions = cl_parameters.GetOptions();
+    ConstitutiveLawOptions.Set(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN, true);
+    ConstitutiveLawOptions.Set(ConstitutiveLaw::COMPUTE_STRESS, true);
+    ConstitutiveLawOptions.Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR, false);
+
+    cl_parameters.SetElementGeometry(Geom);
+    cl_parameters.SetProcessInfo(test_model_part.GetProcessInfo());
+    cl_parameters.SetMaterialProperties(material_properties);
+    cl_parameters.SetStrainVector(strain_vector);
+    cl_parameters.SetStressVector(stress_vector);
+    Matrix const_matrix;
+    cl_parameters.SetConstitutiveMatrix(const_matrix);
+
+    // Create the CL's
+    GenericSmallStrainHighCycleFatigueLaw<GenericConstitutiveLawIntegratorDamage<VonMisesYieldSurface<ModifiedMohrCoulombPlasticPotential<6>>>> HCF_CL = GenericSmallStrainHighCycleFatigueLaw<GenericConstitutiveLawIntegratorDamage<VonMisesYieldSurface<ModifiedMohrCoulombPlasticPotential<6>>>>();
+
+    HCF_CL.SetFatigueReductionFactor(8.8914e-01);
+    Vector previous_stresses=ZeroVector(2);
+    previous_stresses[0]=7.6380e+08;
+    previous_stresses[1]=6.9760e+08;
+    
+    HCF_CL.SetPreviousStresses(previous_stresses);
+    HCF_CL.SetMaxStress(0.0);
+    HCF_CL.SetMinStress(0.0);
+    HCF_CL.SetPreviousMaxStress(7.6380e+08);
+    HCF_CL.SetPreviousMinStress(-2.2914e+08);
+    HCF_CL.SetNumberOfCycles(984);
+    HCF_CL.SetReversionFactor(-0.3);
+    HCF_CL.SetCycleCounter(false);
+    HCF_CL.SetFatigueReductionParameter(5.3239E-07);
```

Comment:
is this the standard way to set parameters in the CLaw? I thought we use parameters for this?

Reply:
These are not parameters, they are internal variables and thus we use private methods to set their values for the test.

Reply:
hm then you discovered a fundamental problem of testing :)
So far we did not do this bcs it opens up the interface of a class for "misuse"

@RiccardoRossi what do you think abt this?